File: lib/data/isar/models/player.dart
import 'package:isar/isar.dart';

part 'player.g.dart';

enum Skill {
  strong,
  developing,
}

@collection
class Player {
  Id id = Isar.autoIncrement;

  @Index(unique: true)
  late String uuid;

  late String name;

  @enumerated
  late Skill skill;

  /// Team UUID this player belongs to (for multi-coach sync).
  String? teamId;

  late DateTime createdAt;

  late DateTime updatedAt;
  String? updatedBy;
  DateTime? deletedAt;
  late int schemaVersion;

  Player();

  Player.create({
    required this.uuid,
    required this.name,
    this.skill = Skill.developing,
    this.teamId,
    DateTime? createdAt,
    DateTime? updatedAt,
    this.updatedBy,
    this.deletedAt,
    int schemaVersion = 1,
  })  : createdAt = createdAt ?? DateTime.now(),
        updatedAt = updatedAt ?? DateTime.now(),
        schemaVersion = schemaVersion;
}

File: backend/src/models/Player.js
const mongoose = require('mongoose');
const { v4: uuidv4 } = require('uuid');

const playerSchema = new mongoose.Schema(
  {
    uuid: { type: String, required: true, unique: true, default: () => uuidv4() },
    teamId: { type: String, required: true },
    name: { type: String, required: true },
    skill: { type: String, required: true, enum: ['strong', 'developing'], default: 'developing' },
    createdAt: { type: Date, default: () => new Date() },
    updatedAt: { type: Date, default: () => new Date() },
    updatedBy: { type: String, default: null },
    deletedAt: { type: Date, default: null },
    schemaVersion: { type: Number, default: 1 },
  },
  { timestamps: false }
);

playerSchema.index({ teamId: 1, updatedAt: 1 });

const Player = mongoose.model('Player', playerSchema);
module.exports = { Player };
